src/output/plugins/RecorderOutputPlugin.cxx

```diff
@@ -200,9 +268,78 @@ RecorderOutput::Commit()
 		throw;
 	}
 
+	/* move file to archive if requested */
+	if (archive_requested && can_archive) {
+
+		// wait for previous move operation to finish
+		if (archive_thread.IsDefined())
+			archive_thread.Join();
+
+		// run move/copy asynchronously to not block output operation
+		//
+		archive_source = strdup(path.c_str());
+		archive_dest = strdup(archive_path.c_str());
+
+		archive_thread.Start();
+
+		archive_requested = 0;
+	}
+	else
+	{
+	    /* delete file if requested */
+	    if (delete_after_record) {
+		    FormatDebug(recorder_domain, "Deleting \"%s\"", path.c_str());
+		    
+		    unlink(path.c_str());
+	    }
+	}
+
 	delete file;
 }
 
+void
+RecorderOutput::ArchiveTask() {
+	int rc;
+
+	SetThreadName("archive_file");
+
+	if (!archive_source || !archive_dest)
+		return;
+
+	if (delete_after_record) {
+		rc = rename (archive_source, archive_dest);
+	} else {
+		try {
+			std::ifstream  src(archive_source, std::ios::binary);
+			std::ofstream  dst(archive_dest,   std::ios::binary);
+
+			dst << src.rdbuf();
+
+			rc = 0;
+		} 
+		catch(...)
+		{
+			rc = -1;
+		}
+	}
+
+	if (rc == 0) {
+		FormatDebug(recorder_domain, "%s \"%s\" to \"%s\"",
+			delete_after_record ? "Moved" : "Copied",
+			archive_source, archive_dest);
+	} else {
+		FormatError(recorder_domain, "Failed to %s \"%s\" to \"%s\", rc=%d",
+			delete_after_record ? "move" : "copy",
+			archive_source, archive_dest, errno);
```

Comment:
`errno` isn't portable. Maybe log the exception you caught, instead of using errno?